subprojects/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonFactory.java

```diff
@@ -26,6 +28,7 @@
  * Controls the lifecycle of the worker daemon and provides access to it.
  */
 @ThreadSafe
+@ServiceScope(Scopes.BuildTree)
```

Comment:
@ghale Do you see this as an issue? Moving the scope of this factory from build session to build tree?

Reply:
Actually I think it's moving from project scope to build tree.  This should be fine.  Seems like you could even move it up to session scope if you want.  It doesn't retain any state or anything - everything is tied up in the clients manager.  There were historical reasons why this was at project scope that have since gone away.

Reply:
Yes, moving up is in general ok because it means less resource consumption and more sharing. But it might mean more leaking of resources too.